x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/predicate/operator/fulltext/MultiMatchQuerySerializationTests.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.esql.expression.predicate.operator.fulltext;
+
+import org.elasticsearch.xpack.esql.core.expression.predicate.fulltext.MultiMatchQueryPredicate;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class MultiMatchQuerySerializationTests extends AbstractFulltextSerializationTests<MultiMatchQueryPredicate> {
+
+    private static final String COMMA = ",";
+
+    @Override
+    protected final MultiMatchQueryPredicate createTestInstance() {
+        return new MultiMatchQueryPredicate(
+            randomSource(),
+            randomFieldString(),
+            randomAlphaOfLength(randomIntBetween(1, 16)),
+            randomOptionOrNull()
+        );
+    }
+
+    @Override
+    protected MultiMatchQueryPredicate mutateInstance(MultiMatchQueryPredicate instance) throws IOException {
+        var fieldString = instance.fieldString();
+        var query = instance.query();
+        var options = instance.options();
+        switch (randomFrom(List.of(1, 2, 3))) {
```

Comment:
I usually use `between(0, 2)` for this. I wonder if this is more clear.... I dunno!